setup.py

```diff
@@ -62,6 +62,7 @@ def maybe_require(pkg):
 
 develop = set([
     'wheel',
+    'rhsm',
```

Comment:
@ahitacat Since this is for the client specifically could you please create a client-develop set with only rhsm in it. Then on line 134/135 add the client-develop in the client-develop extra requires. Also could you elaborate on why this is needed, I'm pretty sure rhsm should be pulled in via the subscription-manager-rhsm rpm, and not the module from pypi but I could be wrong.

Reply:
Hi!
I found that if you install the client within the virtualenv for development, the rhsm is not pulled and thus you need it in the environment. And sure, I can add it in a client-develop set.

Reply:
If I’m not mistaken things installed from RPMs are only available for the system-wide Python and not in virtual environments, as @ahitacat says.

Reply:
@ryan-blakley _rhsm_ is actually imported in [_insights-core_](https://github.com/RedHatInsights/insights-core/blob/e6b64c3c02752a737e61123f9167903cc058c5dd/insights/client/auto_config.py#L102) on auto config. I’d say it’s essential for it to even run and not limited to _insights-client_ development.

Reply:
@ahitacat @Glutexo you are correct it is in insights-core but it's only imported by the client code in insights-core, and only necessary for client development/testing. I think it makes more sense to include this in the existing `develop-client` set or create a separate set as @ryan-blakley suggested to include that module.  We have several teams that use insights-core but not the client so it seems to make things cleaner and more obvious.

Reply:
@bfahr I’d put that into the _client_ section, because it’s required to even _run_ the client in its own virtual environment. If I am not mistaken, when used from the installed RPM package as an egg, the dependencies are not installed by _setuptools_, but by RPM, so there should be no interference.

Reply:
Adding it to the client set works also if it's required for the client to run. In regards to what I mentioned about the rpm, you can create virtualenv's with the --system-site-packages parameter that gives the venv access to the system's module directory. I mentioned the rpm only because the version of rhsm on pypi is older than what's shipped in rhel, and it doesn't appear to really be maintained as there is only two version, but I could be wrong. So I thought I'd mention the rpm in case there was differences in the code between the versions.

Reply:
@ryan-blakley Is there a reason for _rhsm_ in PyPI not being updated?

Reply:
@Glutexo I personally don't know I'm not involved with rhsm in anyway I'm on the insights-core team. I just checked pypi to make sure the package existed initially, and saw it only had two version both of which are older than what's in rhel. But taking another look at the pypi project it's maintained by someone outside of Red Hat, so I'm now less inclined to include that requirement. That might be why it hasn't been a requirement to date.

Reply:
@ryan-blakley Interesting! Maybe grabbing that from PyPI is not the best thing to do and we should rather find a way how to access the RPM library from the virtual environment. I think Chris would be able to help with that. Thank you, Ryan!

Reply:
@Glutexo No problem, you can access the rpm library by creating a virtualenv with the following command. `$ python3 -m venv --system-site-packages .venv` I tested this and was able to import rhsm from within the freshly created venv.